From e40ee37409aff7fcc53a6045620c821a168a8282 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 19 Jan 2024 09:20:02 +0530
Subject: [PATCH 0921/1141] arm64: dts: st: Add node for on-board hub connected
 on stm32mp257f-ev1

MP25-Eval board has on-board hub connected to the usbh-usb2phy1
interface

Change-Id: I11f40aa7b3c58b78c78197d9426b855d9389d456
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/352606
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
index add5615244d5..d03a69297b73 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -701,6 +701,17 @@ &usb2_phy2 {
 &usbh {
 	status = "okay";
 
+	usbh_ehci: usb@482f0000 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		/* onboard HUB */
+		hub@1 {
+			compatible = "usb424,2514";
+			reg = <1>;
+			vdd-supply = <&scmi_v3v3>;
+		};
+	};
+
 	usbh_ohci: usb@482e0000 {
 		status = "disabled";
 	};
-- 
2.39.2

